### PR TITLE
[luci] Support shape inference of ArgMin/Max with negative axis

### PR DESCRIPTION
**Pull request by jinevening**
This supports shape inference of ArgMin/Max with negative axis.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

### CODE REVIEW

File: compiler/luci/service/src/CircleShapeInferenceRule.cpp [hunk 1]
```diff
@@ -279,8 +279,9 @@ template <class CIRCLENODE> loco::NodeShape infer_arg_maxmin(const CIRCLENODE *n
 
     select_axis = const_shape_node->template scalar<loco::DataType::S32>();
   }
-  assert(select_axis < input_shape.rank());
-  assert(select_axis >= 0); // TODO support minus of this breaks
```
Comment (jinevening) [inline compiler/luci/service/src/CircleShapeInferenceRule.cpp:283]: This PR removes this TODO comment

**Review by seanshpark — APPROVED** [review]
LGTM

File: compiler/luci/service/src/CircleShapeInferenceRule.cpp [hunk 2]
```diff
@@ -279,8 +279,9 @@ template <class CIRCLENODE> loco::NodeShape infer_arg_maxmin(const CIRCLENODE *n
 
     select_axis = const_shape_node->template scalar<loco::DataType::S32>();
   }
-  assert(select_axis < input_shape.rank());
```
Comment (seanshpark) [inline compiler/luci/service/src/CircleShapeInferenceRule.cpp:282]: maybe this line is valid?

File: compiler/luci/service/src/CircleShapeInferenceRule.cpp (hunk 1 above)
Comment (jinevening) [reply compiler/luci/service/src/CircleShapeInferenceRule.cpp:283]: I'll upload a recipe for a negative axis in the following PR.

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**